libs/coin-modules/coin-evm/src/__tests__/coin-tester/scenarios/ethereum.ts

```diff
@@ -0,0 +1,263 @@
+  USDC_ON_ETHEREUM,
+  ERC721Interface,
+  ERC1155Interface,
+} from "../helpers";
+import { clearExplorerAppendix, getLogs, setBlock } from "../indexer";
+import { killAnvil, spawnAnvil } from "../anvil";
+
+const makeScenarioTransactions = ({
+  address,
+}: {
+  address: string;
+}): ScenarioTransaction<EvmTransaction>[] => {
+  const scenarioSendEthTransaction: ScenarioTransaction<EvmTransaction> = {
+    name: "Send ethereum",
+    amount: new BigNumber(100),
+    recipient: "0x6bfD74C0996F269Bcece59191EFf667b3dFD73b9",
+    expect: (previousAccount, currentAccount) => {
+      const [latestOperation] = currentAccount.operations;
+      expect(currentAccount.operations.length - previousAccount.operations.length).toBe(1);
+      expect(latestOperation.type).toBe("OUT");
+      expect(latestOperation.value.toFixed()).toBe(latestOperation.fee.plus(100).toFixed());
+      expect(currentAccount.balance.toFixed()).toBe(
+        previousAccount.balance.minus(latestOperation.value).toFixed(),
+      );
+    },
+  };
+
+  const scenarioSendUSDCTransaction: ScenarioTransaction<EvmTransaction> = {
+    name: "Send USDC",
+    amount: new BigNumber(80),
+    recipient: "0x6bfD74C0996F269Bcece59191EFf667b3dFD73b9",
+    subAccountId: encodeTokenAccountId(`js:2:ethereum:${address}:`, USDC_ON_ETHEREUM),
+    expect: (previousAccount, currentAccount) => {
+      const [latestOperation] = currentAccount.operations;
+      expect(currentAccount.operations.length - previousAccount.operations.length).toBe(1);
+      expect(latestOperation.type).toBe("FEES");
+      expect(latestOperation.value.toFixed()).toBe(latestOperation.fee.toFixed());
+      expect(latestOperation.subOperations?.[0].type).toBe("OUT");
+      expect(latestOperation.subOperations?.[0].value.toFixed()).toBe("80");
+    },
+  };
+
+  const scenarioSendERC721Transaction: ScenarioTransaction<EvmTransaction & EvmNftTransaction> = {
+    name: "Send ERC721",
+    recipient: "0x6bfD74C0996F269Bcece59191EFf667b3dFD73b9",
+    mode: "erc721",
+    nft: {
+      tokenId: "3368",
+      contract: "0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D",
+      quantity: new BigNumber(1),
+      collectionName: "Bored Ape",
+    },
+    expect: (previousAccount, currentAccount) => {
+      const [latestOperation] = currentAccount.operations;
+      expect(currentAccount.operations.length - previousAccount.operations.length).toBe(1);
+      expect(
+        currentAccount.nfts?.every(
+          nft => nft.contract !== "0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D",
+        ),
+      ).toBe(true);
+      expect(latestOperation.type).toBe("FEES");
+      expect(
+        latestOperation?.nftOperations?.find(
+          op => op.contract === "0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D",
+        ),
+      ).toBeDefined();
+      expect(
+        latestOperation.nftOperations?.find(
+          operation => operation.contract === "0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D",
+        ),
+      ).toBeDefined();
```

Comment:
Loops are probably overkill, `const lastNftOperation = lastOperation?.nftOperations[0]` and you can expect it to equal an `expect.objectContaining({ contract: "0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D", tokenId: "3368", value: new BigNumber(1) })`